models/klaviyo__events.sql

```diff
@@ -0,0 +1,73 @@
+with events as (
+
+    {% set exclude_fields = ['campaign_or_flow_partition', 'last_touch_campaign_or_flow_id'] %}
+    {% set exclude_fields = exclude_fields | upper if target.type == 'snowflake' else exclude_fields %}
+
+    select 
+        {{ dbt_utils.star(from=ref('int_klaviyo__event_attribution'), except=exclude_fields) }},
+
+        case 
+            when last_touch_type = 'campaign' then last_touch_campaign_or_flow_id 
+        else null end as last_touch_campaign_id,
+        case 
+            when last_touch_type = 'flow' then last_touch_campaign_or_flow_id 
+        else null end as last_touch_flow_id
+
+    from {{ ref('int_klaviyo__event_attribution') }}
+),
+
+campaign as (
+
+    select *
+    from {{ var('campaign') }}
+),
+
+flow as (
+
+    select *
+    from {{ var('flow') }}
+),
+
+person as (
+
+    select *
+    from {{ var('person') }}
+),
+
+metric as (
+
+    select *
+    from {{ var('metric') }}
+),
+
+integration as (
+
+    select *
+    from {{ var('integration') }}
+),
+
+join_fields as (
+
+    select
+        events.*,
+        campaign.campaign_name,
```

Comment:
I think it would be clearer here to have `campaign.campaign_name as last_touch_campaign_name`

Reply:
interested to hear your thoughts